### PR TITLE
MOB-108 Implement score formula

### PR DESCRIPTION
**Pull request by Dezash**
https://kamandax.atlassian.net/browse/MOB-108

Formula: https://github.com/KamandaX/Wiki/blob/master/Score_Calculation.md

### CODE REVIEW

File: API/Services/ScoreService.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using System.Collections.Generic;
+
+namespace API.Services
+{
+    public class ScoreService : IScoreService
+    {
+        public IDictionary<int, float> GetAspectScores(IList<QuestionOption> questionOptions)
+        {
+            IDictionary<int, float> aspectScores = new Dictionary<int, float>();
+            IDictionary<int, int> aspectCounts = new Dictionary<int, int>();
```
Comment (wizacass) [inline API/Services/ScoreService.cs:12]: perhaps we should use `var` here?

File: API/Services/ScoreService.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+            foreach (QuestionOption questionOption in questionOptions)
```
Comment (wizacass) [inline API/Services/ScoreService.cs:14]: Perhaps we should use `var` inside foreach?

File: API/Services/ScoreService.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            foreach (int aspectID in aspectCounts.Keys)
+                aspectScores[aspectID] /= aspectCounts[aspectID];
+
+            return aspectScores;
+        }
+
+        public double GetProductScore(IDictionary<int, float> aspectScores, Score weight)
+        {
+            if (weight == null)
+                throw new ArgumentNullException("weight", "Object instance is not set");
```
Comment (wizacass) [inline API/Services/ScoreService.cs:41]: I think we should avoid using hardcoded string for variable here and instead use `nameof(weight)`. This allows us to better handle refactors as we wouldn't need to search for hardcoded variable names.

File: API/Services/ScoreService.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+            double score = 0;
+            foreach (KeyValuePair<int, float> aspectScore in aspectScores)
+                score += aspectScore.Value * weight.GetScoreByID(aspectScore.Key);
```
Comment (wizacass) [inline API/Services/ScoreService.cs:45]: Maybe it is possible to use LINQ here? Something like 
```C#
double score = aspectScores.Aggregate<KeyValuePair<int, float>, double>(0, (current, aspectScore) => 
current + aspectScore.Value * weight.GetScoreByID(aspectScore.Key));
```

File: API/Services/ScoreService.cs (hunk 1 above)
Comment (Dezash) [reply API/Services/ScoreService.cs]: Done

File: API/Services/ScoreService.cs (hunk 2 above)
Comment (Dezash) [reply API/Services/ScoreService.cs]: Done

File: API/Services/ScoreService.cs (hunk 3 above)
Comment (Dezash) [reply API/Services/ScoreService.cs]: Done

File: API/Services/ScoreService.cs (hunk 4 above)
Comment (Dezash) [reply API/Services/ScoreService.cs]: Done

**Pull request merged**